conan/tools/files/files.py

```diff
@@ -341,18 +342,24 @@ def print_progress(_, __):
                 except Exception as e:
                     output.error("Error extract %s\n%s" % (file_.filename, str(e)))
         else:  # duplicated for, to avoid a platform check for each zipped file
+            file_timestamps = []
             for file_ in zip_info:
                 extracted_size += file_.file_size
                 print_progress(extracted_size, uncompress_size)
                 try:
                     z.extract(file_, full_path)
+                    file_path = os.path.join(full_path, file_.filename)
+                    ts = time.mktime(file_.date_time + (0, 0, -1))
+                    file_timestamps.append((file_path, ts))
                     if keep_permissions:
                         # Could be dangerous if the ZIP has been created in a non nix system
                         # https://bugs.python.org/issue15795
                         perm = file_.external_attr >> 16 & 0xFFF
-                        os.chmod(os.path.join(full_path, file_.filename), perm)
+                        os.chmod(file_path, perm)
                 except Exception as e:
                     output.error("Error extract %s\n%s" % (file_.filename, str(e)))
+            for file_path, ts in file_timestamps:
+                os.utime(file_path, (ts, ts))
```

Comment:
Why the ``z.extract()`` will not automatically handle the zip timestamp? It doesn't make much sense that this has to be handled manually?

Reply:
[Beats me](https://stackoverflow.com/q/9813243). Seems like quite the oversight for the API.

Apparently not even `.extractall()` gets you that.